### PR TITLE
[Draft][luci/pass]Introduce ChangePackToReshape on luci/pass

### PR DESCRIPTION
**Pull request by struss**
For #5180 

This commit is draft for `ChangePackToReshape`.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci/pass/src/SubstitutePackToReshapePass.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  if (target_node == nullptr)
+    return false;
+  if (target_node->values_count() != 1)
+    return false;
+  auto value_node = loco::must_cast<luci::CircleNode *>(target_node->values(0));
+  if (value_node->shape_status() != luci::ShapeStatus::VALID)
+    return false;
+  uint32_t axis = static_cast<uint32_t>(target_node->axis());
+  auto graph = target_node->graph();
+  auto reshape_node = graph->nodes()->create<luci::CircleReshape>();
```
Comment (llFreetimell) [inline compiler/luci/pass/src/SubstitutePackToReshapePass.cpp:36]:
```suggestion
  auto reshape_node = graph->nodes()->create<luci::CircleReshape>();
  reshape_node->tensor(value_node);
```

File: compiler/luci/pass/src/SubstitutePackToReshapePass.test.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    {
+      auto pack = dynamic_cast<luci::CirclePack *>(node);
+      if (not pack)
+        continue;
+      pack_node = pack;
+      continue;
+    }
+    reshape_node = reshape;
+    break;
+  }
```
Comment (llFreetimell) [inline compiler/luci/pass/src/SubstitutePackToReshapePass.test.cpp:96]: Is this what you wanted?

```suggestion
  for (auto node : loco::active_nodes(loco::output_nodes(graph.get())))
  {
    if (dynamic_cast<luci::CircleReshape *>(node))
      reshape_node = dynamic_cast<luci::CircleReshape *>(node)
    else if (dynamic_cast<luci::CirclePack *>(node))
      pack_node = dynamic_cast<luci::CirclePack *>(node);
  }
```

File: compiler/luci/pass/src/SubstitutePackToReshapePass.cpp (hunk 1 above)
Comment (struss) [reply compiler/luci/pass/src/SubstitutePackToReshapePass.cpp:36]: Oh I currently working it. sorry for this....

File: compiler/luci/pass/src/SubstitutePackToReshapePass.test.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci/pass/src/SubstitutePackToReshapePass.test.cpp:96]: yes. I will fix it.

File: compiler/luci/pass/src/SubstitutePackToReshapePass.cpp (hunk 1 above)
Comment (struss) [reply compiler/luci/pass/src/SubstitutePackToReshapePass.cpp:36]: Oh i will fix it.

File: compiler/luci/pass/src/SubstitutePackToReshapePass.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+{
+  auto target_node = dynamic_cast<luci::CirclePack *>(node);
+  if (target_node == nullptr)
+    return false;
+  if (target_node->values_count() != 1)
+    return false;
+  auto value_node = loco::must_cast<luci::CircleNode *>(target_node->values(0));
+  if (value_node->shape_status() != luci::ShapeStatus::VALID)
+    return false;
+  uint32_t axis = static_cast<uint32_t>(target_node->axis());
```
Comment (llFreetimell) [inline compiler/luci/pass/src/SubstitutePackToReshapePass.cpp:34]: `target_node->axis()` can be negative values.
For example, if `target_node->axis()` is -1, it means last dimension.

Comment (struss) [reply compiler/luci/pass/src/SubstitutePackToReshapePass.cpp:34]: Oh Thanks, I will consider it.

**Comment by struss** [comment]
All Done! Close!

**Pull request closed without merging**